pkg/utils/cluster.go

```diff
@@ -40,14 +63,14 @@ func k8sSingleNodeClusterStatus(c client.Client) (bool, error) {
 	return false, nil
 }
 
-func openshiftSingleNodeClusterStatus(c client.Client) (bool, error) {
+func openshiftControlPlaneTopologyStatus(c client.Client) (configv1.TopologyMode, error) {
 	infra := &configv1.Infrastructure{}
 	err := c.Get(context.TODO(), types.NamespacedName{Name: infraResourceName}, infra)
 	if err != nil {
-		return false, err
+		return "", err
```

Comment:
Please, add context information to the error:

```suggestion
return "", fmt.Errorf("can't get Infrastructure [%s]: %w, infraResourceName, err)
```

Reply:
updated.